### PR TITLE
Add user authentication

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
I added user authentication to the application using the `devise` gem in this branch.
I added the following devise modules to the User model:
- Database Authenticable
- Registerable
- Recoverable
- Confirmable
- Validatable

I also copied the views from the devise gem to my project to allow for customization and styling.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -31,5 +31,5 @@
     <% end %>
   </ul>
 
-  <%= 'This user has not posted anything yet!' if @user.posts.empty? %>
+  <%= 'This user has not scribbled anything yet!' if @user.posts.empty? %>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/index.html.erb:34]: - [ ] [OPTIONAL] I tried scribbling a new scribble :laughing:  and the scribble threw an error, kindly fix this scribbling app :roll_eyes: 
Screenshot from 2023-06-29 12-51-52

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @preciousbetine , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed
- [ ] devise used correctly
- [ ] nice design

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 1 above)
Comment (preciousuwhubetine) [reply app/views/posts/index.html.erb:34]: @VICTOR-CHUKWU lol thanks. Will look into it.

Comment (preciousuwhubetine) [reply app/views/posts/index.html.erb:34]: The error is raised only when you try to create a post without signing in. I think that's what the next part of the project deals with.

**Pull request merged**